pkg/controller/linstorsatelliteset/linstorsatelliteset_controller.go

```diff
@@ -530,20 +530,43 @@ func (r *ReconcileLinstorSatelliteSet) reconcileAutomaticDeviceSetup(ctx context
 		"Op":        "reconcileAutomaticDeviceSetup",
 	})
 
-	logger.Debug("check for re-used device paths")
+	logger.Debug("get existing storage pools on node")
+
+	cached := true
+
+	existingPools, err := linstorClient.Nodes.GetStoragePools(ctx, pod.Spec.NodeName, &lapi.ListOpts{Cached: &cached})
+	if err != nil {
+		return fmt.Errorf("failed to list existing storage pools: %w", err)
+	}
+
+	logger.Debug("check for re-used device paths or existing storage pools")
 
 	devsToConfigure := sets.NewString()
 
+outer:
 	for _, poolConfig := range satelliteSet.Spec.StoragePools.AllPhysicalStorageCreators() {
 		if devsToConfigure.HasAny(poolConfig.GetDevicePaths()...) {
 			return fmt.Errorf("a device referenced in the storage pools is referenced twice")
 		}
 
+		for i := range existingPools {
+			if existingPools[i].SharedSpace == poolConfig.GetName() {
```

Comment:
Is `SharedSpace` what we want to check here? That is intended as a special feature when multiple pools are actually backed by the same physical storage.

Reply:
Good catch. Should be `StoragePoolName` obviously.